### PR TITLE
[WFLY-9107] IIOP security configuration and tests refactor

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.jboss.org/projects/WFLY/issues/WFLY-9107

### CODE REVIEW

**Comment by tadamski** [comment]
This PR is the fix for https://github.com/wildfly/wildfly/pull/10343.

**Comment by bstansberry** [comment]
@tadamski This needs a rebase.

**Comment by kabir** [comment]
@tadamski Needs a rebase

**Comment by tadamski** [comment]
@kabir I have some security errors after rebase - seem not-trivial  - I will ping you when fixed

**Comment by tadamski** [comment]
retest this please

**Comment by tadamski** [comment]
@kabir - fixed, he test that fails on Windows (JspTagTestCase) is unrelated

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
Retest this please

**Comment by stuartwdouglas** [comment]
Retest this please

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPSubsystemAdd.java [hunk 1]
```diff
@@ -194,10 +194,19 @@ public void execute(DeploymentProcessorTarget processorTarget) {
             builder.addDependency(authContextServiceName);
         }
 
+        final boolean serverRequiresSsl = IIOPRootDefinition.SERVER_REQUIRES_SSL.resolveModelAttribute(context, model).asBoolean();
+
         // inject the socket bindings that specify IIOP and IIOP/SSL ports.
         String socketBinding = props.getProperty(Constants.ORB_SOCKET_BINDING);
-        builder.addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBinding), SocketBinding.class,
-                orbService.getIIOPSocketBindingInjector());
+        if (socketBinding != null) {
```
Comment (stuartwdouglas) [inline iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPSubsystemAdd.java]: Can socketBinding actually every be null? Looking at the resource definition it appears to have a default value, which means there will always be a WARN message if server-requires-ssl is set.

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
@tadamski Tests fail

**Comment by rhusar** [comment]
@tadamski Please comply with our minimal requirements for PRs, like links, title, etc.

**Comment by tadamski** [comment]
retest this please

**Comment by tadamski** [comment]
@kabir - fixed the failing test
@rhusar - modified the title and description

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java [hunk 2]
```diff
@@ -78,7 +78,7 @@
 
     protected static final AttributeDefinition SOCKET_BINDING = new SimpleAttributeDefinitionBuilder(
             Constants.ORB_SOCKET_BINDING, ModelType.STRING, true).setAttributeGroup(Constants.ORB)
-            .setDefaultValue(new ModelNode().set("iiop")).setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
+            .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
```
Comment (kabir) [inline iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java]: As we are changing the model, we need a model version bump

**Comment by tadamski** [comment]
I have added the model update.

**Comment by tadamski** [comment]
retest this please

**Comment by tadamski** [comment]
Linux - elytron failures seem unrelated

**Review by jamezp — APPROVED** [review]

**Pull request merged**